### PR TITLE
[4.0] Replace highlighter javascript with mark.js

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
Replace Joomla's own implementation of highlighting with mark.js

### Testing Instructions
Check the smart search in the front end, you should get:
screen shot 2017-05-30 at 12 52 47

### CODE REVIEW

**Comment by brianteeman** [comment]
As this new script has several useful options would it be possible to add them to the plugin configuration options - i can see several of them being very useful and would address issues that have been raised in the past. especially the diacritics and case sensitive

 ```
  "exclude": [],
   "separateWordSearch": true,
   "accuracy": "partially",
   "diacritics": true,
    "synonyms": {},
    "iframes": false,
    "iframesTimeout": 5000,
    "acrossElements": true,
    "caseSensitive": false,
    "ignoreJoiners": false,
   "wildcards": "disabled",
```

**Comment by dgt41** [comment]
@brianteeman sure! will do that

**Comment by brianteeman** [comment]
Not sure how relevant all those options are but some of them definitely are

File: plugins/system/highlight/highlight.php [hunk 1]
```diff
@@ -36,6 +36,7 @@ public function onAfterDispatch()
 		}
 
 		// Set the variables.
+		$doc = JFactory::getDocument();
```
Comment (yvesh) [inline plugins/system/highlight/highlight.php]: cs

**Comment by yvesh** [comment]
You plan to replace the JavaScript tests you deleted? Don't you? ;-)

**Pull request closed without merging**